resolver/resolver.go

```diff
@@ -86,9 +86,15 @@ type Address struct {
 	// Type is the type of this address.
 	Type AddressType
 	// ServerName is the name of this address.
+	// If non-empty, the ServerName is used as the transport certification authority for.
+	// the address.
```

Comment:
Let's also add: ", instead of the hostname from the Dial target string.  In most cases, this should not be set."